frontend/src/common/components/form/subFormComponents/NumberInput.tsx

```diff
@@ -0,0 +1,43 @@
+import { OutlinedInput } from "@mui/material";
+import { useFormContext } from "react-hook-form";
+import { ORBC_FormTypes } from "../../../types/common";
+import { CustomOutlinedInputProps } from "./CustomOutlinedInput";
+import "./NumberInput.scss";
+
+/**
+ * An onRouteBC customized MUI OutlineInput component
+ * that automatically filters out non-numeric values as the user types
+ */
+export const NumberInput = <T extends ORBC_FormTypes>(
```

Comment:
So what exactly is the purpose of this new component? To only allow users to enter numeric inputs? Should we replace the year input with this as well then?

Reply:
The purpose if the NumberInput component is as you mentioned, to prevent users from entering non numeric characters. If we have no need for the stepper on the vehicle year input then we could opt to replace it with this, however the [update ticket](https://moti-imb.atlassian.net/browse/ORV2-2939) does not specify this.

Reply:
ok no worries